pkg/kademlia/kademlia_test.go

```diff
@@ -30,6 +31,10 @@ func kadconfig() KadConfig {
 }
 
 func TestNewKademlia(t *testing.T) {
+	dir, err := ioutil.TempDir(os.TempDir(), "test")
```

Comment:
Also, you don't need to provide the `os.TempDir`. 

```
dir, err := ioutil.TempDir("", "kademlia")
```

Is sufficient. (same for all other locations)